File: HappyNumber.cpp
class Solution {
public:
    bool isHappy(int n) {
        int slow = n;
        int fast = n;
        do{
            slow=sum(slow);
            fast=sum(sum(fast));
            if(slow==1)
                return true;
            }
            while(slow!=fast);
                return false;
            
            
        }
    
    private:
    int sum(int n){
        int sum =0;
         while(n>0){
         int rem = n%10;
         sum+=(rem*rem);
         n=n/10;
         }
         return sum;
    }
    
};

File: HappyNumber.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        while (n not in seen):
            seen.add(n)
            new = 0
            for ele in str(n):
                new += int(ele) ** 2
                
            
            if new == 1: return True
            n = new
        return False
